packages/ts/generator-plugin-signals/src/SignalProcessor.ts

```diff
@@ -0,0 +1,132 @@
+        transform<SourceFile>((tsNode) => {
+          if (ts.isFunctionDeclaration(tsNode) && tsNode.name?.text === method.name) {
+            const endpointAndMethodName = ts.factory.createStringLiteral(`${this.#service}.${method.name}`);
+            const body = template(
+              `
+function dummy() {
+  return new ${NUMBER_SIGNAL_CHANNEL}(${ENDPOINT_AND_METHOD_NAME}, ${CONNECT_CLIENT}).signal;
+}`,
+              (statements) => (statements[0] as FunctionDeclaration).body?.statements,
+              [
+                transform((node) =>
+                  ts.isIdentifier(node) && node.text === ENDPOINT_AND_METHOD_NAME ? endpointAndMethodName : node,
+                ),
+                transform((node) =>
+                  ts.isIdentifier(node) && node.text === NUMBER_SIGNAL_CHANNEL ? numberSignalChannelId : node,
+                ),
+                transform((node) => (ts.isIdentifier(node) && node.text === CONNECT_CLIENT ? connectClientId : node)),
+              ],
+            );
+
+            let returnType = tsNode.type;
+            if (
+              returnType &&
+              ts.isTypeReferenceNode(returnType) &&
+              'text' in returnType.typeName &&
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
+              returnType.typeName.escapedText === 'Promise'
+            ) {
+              if (returnType.typeArguments && returnType.typeArguments.length > 0) {
+                returnType = returnType.typeArguments[0];
+              }
+            }
+
+            return ts.factory.createFunctionDeclaration(
+              tsNode.modifiers?.filter((modifier) => modifier.kind !== ts.SyntaxKind.AsyncKeyword),
+              tsNode.asteriskToken,
+              tsNode.name,
+              tsNode.typeParameters,
+              [],
+              returnType,
+              ts.factory.createBlock(body ?? [], false),
+            );
+          }
+
+          return tsNode;
+        }),
+      ),
+    ]).transformed;
+
+    this.#removeUnusedRequestInitImports(file);
+
+    return createSourceFile(
+      [
+        ...this.#dependencyManager.imports.toCode(),
+        ...file.statements.filter((statement) => !ts.isImportDeclaration(statement)),
+      ],
+      file.fileName,
+    );
+  }
+
+  #processSignalImports(signalImports: string[]) {
+    const { imports } = this.#dependencyManager;
+
+    signalImports.forEach((signalImport) => {
+      const result = imports.default.find((p) => p.includes(signalImport));
+
+      if (result) {
+        const [path, _, id] = result;
+        imports.default.remove(path);
+        imports.named.add(HILLA_REACT_SIGNALS, id.text, true, id);
+      }
+    });
+  }
+
+  #removeUnusedRequestInitImports(file: SourceFile) {
+    const transformedFileText = ts.createPrinter().printFile(file);
+
+    const hasNormalEndpointCalls =
+      transformedFileText.includes('init?: EndpointRequestInit') && transformedFileText.includes('.call(');
```

Comment:
Do I understand correctly that you're checking if the `init` is used here?

Reply:
Yeah, that's the idea, since our generated services with calls to normal endpoints will have `init?: EndpointRequestInit_1` (or some other _N), but services without normal endpoint, will not have methods with that parameter after transformation.

Reply:
Maybe, it'd be better to check the function's argument to be used in the function body? Something like

```ts
transform((node) => {
  if (
    ts.isFunctionDeclaration(node) &&
    node.parameters.some(({ name }) => ts.isIdentifier(name) && name.text === 'init') &&
    node.body
  ) {
    let hasInit = false;

    traverse(node.body, (n) => {
      if (ts.isIdentifier(n) && n.text === 'init') {
        hasInit = true;
      }
    });

    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
    return hasInit
      ? node
      : ts.factory.createFunctionDeclaration(
          node.modifiers,
          node.asteriskToken,
          node.name,
          node.typeParameters,
          node.parameters.filter(({ name }) => !(ts.isIdentifier(name) && name.text === 'init')),
          node.type,
          node.body,
        );
  }

  return node;
}),
```
Where `traverse` function should be defined in `generator-utils/ast.ts` as:
```ts
export function traverse(root: Node, visitor: (node: Node) => void): void {
  const _visitor = (node: Node) => {
    visitor(node);
    ts.forEachChild(node, _visitor);
  };

  ts.forEachChild(root, _visitor);
}
```

Reply:
Well, if I understand this correctly, this is a nice improvement for the part I've just passed `[]` (an empty array) to the `createFunctionDeclaration` as the `parameters`, correct? Ofc, I can use it with another variable scoped outside of this transformer to detect whether the `init` is used in any of the functions or not.

Reply:
ah, if you don't need any parameters, then sure. I just focused on removing the `init`one.

Reply:
Since the methods that are not returning signal instances are not transformed by this plugin, it's easy just to exclude the `init` parameter for the methods returning the signal instance, as we are sure that is not used in those method bodies.